### ISSUE TITLE
Archived on CRAN: Use paws R Library as dependency

### ISSUE THREAD

**Issue by jtelleriar** [sicarul/redshiftTools#53]
In order to put it back to CRAN, I would use Python's boto3 with reticulate or R's paws Library:

https://cran.r-project.org/web/packages/paws/index.html

**Comment by JakeRuss** [comment]
@sicarul Pablo, do you have plans to continue maintaining this package?

**Comment by sicarul** [comment]
Hey @JakeRuss i'm no longer using Amazon Redshift which complicates mantaining the package for me. If someone wants to take over i'm happy to help with the transition

**Comment by JakeRuss** [comment]
@sicarul I would have interest as I use redshift daily for work. Discussing with my employer whether we'd take over the project. I will reach out to you via email.

**Comment by niklasvm** [comment]
I have used boto3 via reticulate for copying data between local and S3. If you are going to python then I propose using one of the python solutions for uploading and downloading to/from redshift

**Comment by JakeRuss** [comment]
I plan to take over the maintainer-ship of this package soon, unless someone beats me to it. 

My plan is to use AWS' R SDK {paws} to keep the package simpler to implement and maintain. But if someone else takes over before I do and strongly prefers going the {reticulate}/{boto3} route, by all means do so.

**Comment by JakeRuss** [comment]
For anyone wishing to follow my progress, I have started refactoring this package over at https://github.com/RedOakStrategic/redshiftTools/tree/redoakjake. I will have an initial working version soon. @russellpierce

**Comment by jtelleriar** [comment]
Note that aws.s3 package is available again in CRAN, and maintanied by
Simon Urbanek (R Core), so redshiftTools it could be re-uploaded to CRAN.

Best,
Juan

**Comment by JakeRuss** [comment]
I appreciate your note, @jtelleriar. I did see that @su rescued the {aws.s3} package as a temporary measure to get the package back on CRAN, but long term, {paws} is probably a better choice.

**Comment by su** [comment]
Don't know what Su you guys are talking about but it's not me.

Unsubbing now.

**Comment by jtelleriar** [comment]
It's @s-u XD

**Comment by s-u** [comment]
Yes, that was me ;). It depends on your use-cases, if you don't mind having Python dependency with the associated issues (can only embed one runtime, the usual Python version nightmare and foreign syntax) then `paws` has the benefit being able to leverage Amazon's work directly. With `aws.*` you get a native solution without extra dependencies, but the functionality is limited to what we implement in R. The community behind `aws.*` is picking up now but, obviously, we can't match Amazon ;).

**Comment by JakeRuss** [comment]
There is no dependency on python with {paws}. While I applaud you @s-u and the others in the R community for stepping up to rescue {aws.s3}, I am going to press forward with {paws} for the new internals of {redshiftTools}.

**Comment by s-u** [comment]
@JakeRuss oh, sorry, I was reading that it uses `reticulate` to wrap `boto3` - my bad. The API looks like an automated translation. There is pretty much zero information about the project so it's hard to know whom to trust ;). I looked now and I see it is also a wapper of `curl`. Thanks.
